spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java

```diff
@@ -93,7 +95,16 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 
 
 		try {
-			Path workDir = Files.createDirectory(Paths.get(logPathRoot.toFile().getAbsolutePath(),
+			String groupDeploymentId = request.getDeploymentProperties().get(GROUP_DEPLOYMENT_ID);
+			if (groupDeploymentId == null) {
+				groupDeploymentId = request.getDefinition().getGroup() + "-" + System.currentTimeMillis();

```

Comment:
I'd prefer not to have to pass anything for the unit tests, if possible at all. But I see what you're saying trying to do the same calculation here for a bogus group deployment id is wrong. After all we're just trying to create a unique directory if no group deployment id is provided - so perhaps we an just use a timestamp for the folder instead.